service/controller/v2/resource/resourcegroup/resource.go

```diff
@@ -105,21 +107,30 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring resource group is deleted")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring resource group deletion")
 
-	f, err := groupsClient.Delete(ctx, key.ClusterID(customObject))
+	_, err = groupsClient.Get(ctx, key.ClusterID(customObject))
 	if IsNotFound(err) {
 		// fall through
 	} else if err != nil {
 		return microerror.Mask(err)
 	} else {
-		err = f.WaitForCompletion(ctx, groupsClient.Client)
-		if err != nil {
+		_, err := groupsClient.Delete(ctx, key.ClusterID(customObject))
+		if IsNotFound(err) {
+			// fall through
+		} else if err != nil {
 			return microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "resource group deletion initiated")
```

Comment:
Nitpick: I would write `resource group deletion in progress`. It could be initiated in previous reconciliation loop.

Don't we have to check `ProvisioningState` for the group as well? What will happen when you try to delete a group and deletion is in progress?

Reply:
I changed the logging. 

During my testing I saw the deletion being triggered several times until all resources got removed in Azure. `Delete` seems to trigger this properly without conflicts. This was actually my expectation of the API behaviour. On consecutive deletions we just run into this branch of the condition, keep the finalizer, and cancel.

Reply:
Cool.